libraries/AP_InertialSensor/AP_InertialSensor_SCH16T.cpp

```diff
@@ -0,0 +1,448 @@
+static constexpr uint16_t RATE_300DPS_1475HZ = 0b0001001011011011; // Gyro XYZ range 300 deg/s @ 1475Hz
+static constexpr uint16_t ACC12_8G_1475HZ = 0b0001001011011011;  // Acc XYZ range 8 G and 1475 update rate
+static constexpr uint16_t ACC3_26G = (0b000 << 0);
+static constexpr uint16_t SPI_SOFT_RESET = (0b1010);
+static constexpr uint32_t POWER_ON_TIME = 250000UL;
+
+// Data registers
+#define RATE_X1         0x01 // 20 bit
+#define RATE_Y1         0x02 // 20 bit
+#define RATE_Z1         0x03 // 20 bit
+#define ACC_X1          0x04 // 20 bit
+#define ACC_Y1          0x05 // 20 bit
+#define ACC_Z1          0x06 // 20 bit
+#define ACC_X3          0x07 // 20 bit
+#define ACC_Y3          0x08 // 20 bit
+#define ACC_Z3          0x09 // 20 bit
+#define RATE_X2         0x0A // 20 bit
+#define RATE_Y2         0x0B // 20 bit
+#define RATE_Z2         0x0C // 20 bit
+#define ACC_X2          0x0D // 20 bit
+#define ACC_Y2          0x0E // 20 bit
+#define ACC_Z2          0x0F // 20 bit
+#define TEMP            0x10 // 16 bit
+// Status registers
+#define STAT_SUM        0x14 // 16 bit
+#define STAT_SUM_SAT    0x15 // 16 bit
+#define STAT_COM        0x16 // 16 bit
+#define STAT_RATE_COM   0x17 // 16 bit
+#define STAT_RATE_X     0x18 // 16 bit
+#define STAT_RATE_Y     0x19 // 16 bit
+#define STAT_RATE_Z     0x1A // 16 bit
+#define STAT_ACC_X      0x1B // 16 bit
+#define STAT_ACC_Y      0x1C // 16 bit
+#define STAT_ACC_Z      0x1D // 16 bit
+// Control registers
+#define CTRL_FILT_RATE  0x25 // 9 bit
+#define CTRL_FILT_ACC12 0x26 // 9 bit
+#define CTRL_FILT_ACC3  0x27 // 9 bit
+#define CTRL_RATE       0x28 // 15 bit
+#define CTRL_ACC12      0x29 // 15 bit
+#define CTRL_ACC3       0x2A // 3 bit
+#define CTRL_USER_IF    0x33 // 16 bit
+#define CTRL_ST         0x34 // 13 bit
+#define CTRL_MODE       0x35 // 4 bit
+#define CTRL_RESET      0x36 // 4 bit
+// Misc registers
+#define ASIC_ID         0x3B // 12 bit
+#define COMP_ID         0x3C // 16 bit
+#define SN_ID1          0x3D // 16 bit
+#define SN_ID2          0x3E // 16 bit
+#define SN_ID3          0x3F // 16 bit
+
+#define T_STALL_US   20U
+
+#define SPI48_DATA_INT32(a)     (((int32_t)(((a) << 4)  & 0xfffff000UL)) >> 12)
+#define SPI48_DATA_UINT32(a)    ((uint32_t)(((a) >> 8)  & 0x000fffffUL))
+#define SPI48_DATA_UINT16(a)    ((uint16_t)(((a) >> 8)  & 0x0000ffffUL))
+
+extern const AP_HAL::HAL& hal;
+
+AP_InertialSensor_SCH16T::AP_InertialSensor_SCH16T(AP_InertialSensor &imu,
+                                                         AP_HAL::OwnPtr<AP_HAL::Device> _dev,
+                                                         enum Rotation _rotation)
+    : AP_InertialSensor_Backend(imu)
+    , dev(std::move(_dev))
+    , rotation(_rotation)
+{
+    expected_sample_rate_hz = 1475;
+    accel_scale = 1.f / 1600.f;
+    gyro_scale = radians(1.f / 1600.f);
+
+    _registers[0] = RegisterConfig(CTRL_FILT_RATE,  FILTER_68HZ);        // 68Hz -- default
```

Comment:
These are probably too low. Recommend turning off the filter for now.

Reply:
done